src/wazuh_qa_framework/system/wazuh_handler.py

```diff
@@ -918,78 +955,253 @@ def get_node_type(self, host):
         """
         pass
 
     def clean_client_keys(self, hosts=None):
-        """Clean client keys
+        """Clean client keys.
 
         Args:
             hosts (str, optional): Hostname. Defaults to None.
         """
         pass
 
     def clean_agents(self, agents=None):
-        """Stop agents, remove them from manager and clean their client keys
+        """Stop agents, remove them from manager and clean their client keys.
 
         Args:
             agents (_type_, agents_list): Agents list. Defaults to None.
         """
         pass
 
     def remove_agents_from_manager(self, agents=None, status='all', older_than='0s'):
-        """Remove agents from manager
+        """Remove agents from manager.
 
         Args:
             agents (list, optional): Agents list. Defaults to None.
             status (str, optional): Agents status. Defaults to 'all'.
             older_than (str, optional): Older than parameter. Defaults to '0s'.
 
         Returns:
-            dict: API response
+            dict: API response.
         """
         pass
 
     def get_managers(self):
-        """Get environment managers names
+        """Get environment managers names.
 
         Returns:
-            List: Managers names list
+            List: Managers names list.
         """
         return self.get_group_hosts('manager')
 
     def get_agents(self):
-        """Get environment agents names
+        """Get environment agents names.
 
         Returns:
-            List: Agent names list
+            List: Agent names list.
         """
         return self.get_group_hosts('agent')
 
     def is_agent(self, host):
-        """Check if host is agent
+        """Check if host is agent.
 
         Args:
-            host (str): Hostname
+            host (str): Hostname.
         Returns:
-            bool: True if host is agent
+            bool: True if host is agent.
         """
         return host in self.get_agents()
 
     def is_manager(self, host):
-        """Check if host is manager
+        """Check if host is manager.
 
         Args:
-            host (str): Hostname
+            host (str): Hostname.
         Returns:
-            bool: True if host is manager
+            bool: True if host is manager.
         """
         return host in self.get_managers()
 
+    def get_group_list(self, manager, method='api'):
+        """Get list of groups.
+        Args:
+            manager (str): Name of the manager.
+            method (str): Method to be used to get information. Defaults to api.
+        """
+        group_list = []
+        self.logger.info(f'Requesting group list info using {method.upper()}')
+        if method == 'api':
+            endpoint = f'/groups'
+            request = WazuhAPIRequest(endpoint=endpoint, method='GET')
+            for item in request.send(WazuhAPI(address=self.get_host_variables(manager)['ip'])).data['affected_items']:
+                group_list.append(item['name'])
```

Comment:
- Avoid generics variable names
- This `WazuhAPI(address=self.get_host_variables(manager)['ip'])` is used around all the class, consider create a method to get the WazuhAPI class of a manager.

```suggestion
            for group in request.send(WazuhAPI(address=self.get_host_variables(manager)['ip'])).data['affected_items']:
                group_list.append(group['name'])
```

Reply:
Done in https://github.com/wazuh/qa-system-framework/pull/71/commits/56b13fd995d0c2ff5bf72f28f07e8115eb52d246